### PR TITLE
Improvement/sale tunnel ux

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

This PR contains a bunch of little improvement and fix details in the section below.

## Proposal

- [x] Prevent to close sale tunnel modal by mistake
- [x] Do not await logout request succeeded to invalidate session data
- [x] Fix erratic tests failure

### CODE REVIEW

File: src/frontend/js/components/SaleTunnel/index.tsx [hunk 1]
```diff
@@ -100,10 +100,12 @@ const SaleTunnel = ({ product }: SaleTunnelProps) => {
         {product.call_to_action}
       </button>
       <Modal
-        testId="SaleTunnel__modal"
         className="SaleTunnel__modal"
-        onRequestClose={handleModalClose}
         isOpen={isOpen}
+        onRequestClose={handleModalClose}
+        shouldCloseOnOverlayClick={false}
+        shouldCloseOnEsc={false}
```
Comment (sampaccoud) [inline src/frontend/js/components/SaleTunnel/index.tsx:107]: can this be tested?

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ### Changed
 
+- Immediately invalidate session data on logout
```
Comment (sampaccoud) [inline CHANGELOG.md:13]: I don't find this description helpful compared to the commit message. Maybe tell us more about the stakes:
- Avoid lag experienced by user on logout by not waiting for the logout request to succeed

File: src/frontend/js/data/SessionProvider/BaseSessionProvider.tsx [hunk 3]
```diff
@@ -58,7 +58,7 @@ const BaseSessionProvider = ({ children }: PropsWithChildren<any>) => {
   }, [queryClient]);
 
   const destroy = useCallback(async () => {
-    await AuthenticationApi!.logout();
+    AuthenticationApi!.logout();
```
Comment (sampaccoud) [inline src/frontend/js/data/SessionProvider/BaseSessionProvider.tsx:61]: Maybe it should be tested as well?

File: src/frontend/js/components/SaleTunnel/index.tsx (hunk 1 above)
Comment (jbpenrath) [reply src/frontend/js/components/SaleTunnel/index.tsx:107]: Yes

File: src/frontend/js/data/SessionProvider/BaseSessionProvider.tsx (hunk 3 above)
Comment (jbpenrath) [reply src/frontend/js/data/SessionProvider/BaseSessionProvider.tsx:61]: I just have to update a test to not await the destroy method to check if user state has been updated.

File: src/frontend/js/components/SaleTunnel/index.spec.tsx [hunk 4]
```diff
@@ -198,6 +198,14 @@ describe('SaleTunnel', () => {
...
     screen.getByRole('heading', { level: 1, name: 'SaleTunnelStepPayment Component' });
 
+    // - Press the escape key should not close the dialog
+    fireEvent.keyDown(screen.getByTestId('SaleTunnel__modal'), { keyCode: 27 });
+    screen.getByTestId('SaleTunnel__modal');
+
+    // - Click on the overlay area should not close the dialog
+    fireEvent.click(document.querySelector('.modal__overlay')!);
+    screen.getByTestId('SaleTunnel__modal');
+
```
Comment (sampaccoud) [inline src/frontend/js/components/SaleTunnel/index.spec.tsx:208]: :star_struck:

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**